### DISCUSSION TITLE
CPEX45_high_order task list

### DISCUSSION THREAD

**Discussion by jedbrown** [CGNS/CGNS#344] [CGNStalk]
I'm eagerly awaiting this feature and have built the [CPEX_high_order branch](https://github.com/CGNS/CGNS/tree/CPEX45_high_order), but don't know what tasks are remaining or if there are any visualization tools that can work with these files yet (assume not). It would be useful to make a draft pull request with a tasklist or otherwise communicate what remains to be done, and what community testing/contributions are solicited.

**Comment by ScottFowler** [comment]
Here at Tecplot we're currently working on adding support for high order elements, focusing on data import from CGNS first.  The work is currently in development and due for release later this year.  We're not working with files based on CPEX45, but we currently have Tecplot 360 in a state in which is can read all of the high order element types in the current CGNS standard, up to P4.

I'd be interested in learning more about your needs and why CPEX45 is important to you.  That way we can be better prepared to support it moving forward.

Cheers,
Scott Fowler
Tecplot 360 Product Manager
s.fowler@tecplot.com

**Reply by jedbrown** [reply]
Hi, Scott. I'm glad to hear Tecplot is developing this support.

I'm a maintainer of PETSc (and libCEED) looking to improve our CGNS support and provide a robust and scalable checkpoint/visualization experience for high-order unstructured LES/DNS applications (and others). These app users like Paraview, which supports at least cubic elements with CGNS. My reasons for preferring CPEX45 is higher than 4th order and ability to use mixed discretizations (geometry and different fields in different finite element spaces). If CGNS is used as a checkpoint format, it's important to avoid making lossy transformations when writing checkpoints.
